@roigcarlo any idea why this? (only windows)

~~~sh
     Creating library D:/a/Kratos/Kratos/build/Custom/kratos/Custom/Kratos.lib and object D:/a/Kratos/Kratos/build/Custom/kratos/Custom/Kratos.exp
add_other_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: bool __cdecl Kratos::MasterSlaveConstraint::IsActive(void)const " (?IsActive@MasterSlaveConstraint@Kratos@@QEBA_NXZ) referenced in function "public: __cdecl <lambda_05c4e18abb9c086029edb767fe8acad6>::operator()(class Kratos::MasterSlaveConstraint &)const " 
~~~

--- a/kratos/includes/indexed_object.h
+++ b/kratos/includes/indexed_object.h
@@ -4,15 +4,14 @@
 //   _|\_\_|  \__,_|\__|\___/ ____/
 //                   Multi-Physics
 //
-//  License:		 BSD License
-//					 Kratos default license: kratos/license.txt
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
 //
 //  Main authors:    Riccardo Rossi
 //
 //
 
-#if !defined(KRATOS_INDEXED_OBJECT_H_INCLUDED )
-#define  KRATOS_INDEXED_OBJECT_H_INCLUDED
+#pragma once
 
 // System includes
 #include <iostream>
@@ -264,6 +263,4 @@ inline std::ostream& operator << (std::ostream& rOStream,
 ///@}
 
 }  // namespace Kratos.
-
-#endif // KRATOS_INDEXED_OBJECT_H_INCLUDED  defined
 

--- a/kratos/includes/master_slave_constraint.h
+++ b/kratos/includes/master_slave_constraint.h
@@ -71,7 +71,7 @@ namespace Kratos
  * This unique equation is used later on to modify the equation system.
  * @author Aditya Ghantasala
  */
-class MasterSlaveConstraint
+class KRATOS_API(KRATOS_CORE) MasterSlaveConstraint
     :  public IndexedObject, public Flags
 {
 public:

--- a/kratos/processes/integration_values_extrapolation_to_nodes_process.cpp
+++ b/kratos/processes/integration_values_extrapolation_to_nodes_process.cpp
@@ -108,8 +108,7 @@ void IntegrationValuesExtrapolationToNodesProcess::ExecuteFinalizeSolutionStep()
     };
 
     block_for_each(r_elements_array, TLSType(), [&](Element& rElem, TLSType& rTls){
-        // Only active elements. Detect if the element is active or not. If the user did not make any choice the element
-        // NOTE: Is active by default
+        // Only active elements
         if (rElem.IsActive()) {
             auto& r_this_geometry = rElem.GetGeometry();
 
@@ -297,8 +296,7 @@ void IntegrationValuesExtrapolationToNodesProcess::InitializeMaps()
 
     // Fill the average value
     block_for_each(r_elements_array, TLSType(), [&](Element& rElem, TLSType& rTls){
-        // Only active elements. Detect if the element is active or not. If the user did not make any choice the element
-        // NOTE: Is active by default
+        // Only active elements
         if (rElem.IsActive()) {
             // The geometry of the element
             auto& r_this_geometry = rElem.GetGeometry();

--- a/kratos/processes/skin_detection_process.cpp
+++ b/kratos/processes/skin_detection_process.cpp
@@ -76,7 +76,7 @@ void SkinDetectionProcess<TDim>::GenerateFaceMaps(
     for(IndexType i = 0; i < number_of_elements; ++i) {
         auto it_elem = it_elem_begin + i;
 
-        // Detect if the element is active or not. If the user did not make any choice the element is active by default
+        // If the element is active
         if (it_elem->IsActive()) {
             GeometryType& r_geometry = it_elem->GetGeometry();
 
@@ -116,7 +116,7 @@ void SkinDetectionProcess<TDim>::GenerateFaceMaps(
     for(IndexType i = 0; i < number_of_elements; ++i) {
         auto it_elem = it_elem_begin + i;
 
-        // Detect if the element is active or not. If the user did not make any choice the element is active by default
+        // If the element is active
         if (it_elem->IsActive()) {
             GeometryType& r_geometry = it_elem->GetGeometry();
 

--- a/kratos/solving_strategies/builder_and_solvers/explicit_builder.h
+++ b/kratos/solving_strategies/builder_and_solvers/explicit_builder.h
@@ -331,8 +331,7 @@ public:
             // Assemble all elements
             for (int i_elem = 0; i_elem < n_elems; ++i_elem) {
                 auto it_elem = r_elements_array.begin() + i_elem;
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
-                // TODO: We will require to update this as soon as we remove the mIsDefined from the Flags
+                // If the element is active
                 if (it_elem->IsActive()) {
                     // Calculate elemental explicit residual contribution
                     // The explicit builder and solver assumes that the residual contribution is assembled in the REACTION variables
@@ -344,8 +343,7 @@ public:
 #pragma omp for schedule(guided, 512)
             for (int i_cond = 0; i_cond < n_conds; ++i_cond) {
                 auto it_cond = r_conditions_array.begin() + i_cond;
-                // Detect if the condition is active or not. If the user did not make any choice the condition is active by default
-                // TODO: We will require to update this as soon as we remove the mIsDefined from the Flags
+                // If the condition is active
                 if (it_cond->IsActive()) {
                     // Calculate condition explicit residual contribution
                     // The explicit builder and solver assumes that the residual contribution is assembled in the REACTION variables

--- a/kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h
+++ b/kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h
@@ -1244,8 +1244,7 @@ protected:
             #pragma omp for schedule(guided, 512) nowait
             for (int i=0; i<nelements; i++) {
                 typename ElementsArrayType::iterator it = pElements.begin() + i;
-                // Detect if the element is active or not. If the user did not make any choice the element
-                // is active by default
+                // If the element is active
                 if(it->IsActive()) {
                     //calculate elemental Right Hand Side Contribution
                     pScheme->CalculateRHSContribution(*it, RHS_Contribution, EquationId, CurrentProcessInfo);
@@ -1263,8 +1262,7 @@ protected:
             #pragma omp for schedule(guided, 512)
             for (int i = 0; i<nconditions; i++) {
                 auto it = ConditionsArray.begin() + i;
-                // Detect if the element is active or not. If the user did not make any choice the element
-                // is active by default
+                // If the condition is active
                 if(it->IsActive()) {
                     //calculate elemental contribution
                     pScheme->CalculateRHSContribution(*it, RHS_Contribution, EquationId, CurrentProcessInfo);
@@ -1392,8 +1390,7 @@ protected:
             for (int i_const = 0; i_const < number_of_constraints; ++i_const) {
                 auto it_const = rModelPart.MasterSlaveConstraints().begin() + i_const;
 
-                // Detect if the constraint is active or not. If the user did not make any choice the constraint
-                // It is active by default
+                // If the constraint is active
                 if (it_const->IsActive()) {
                     it_const->CalculateLocalSystem(transformation_matrix, constant_vector, r_current_process_info);
                     it_const->EquationIdVector(slave_equation_ids, master_equation_ids, r_current_process_info);

--- a/kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_lagrange_multiplier.h
+++ b/kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_lagrange_multiplier.h
@@ -854,8 +854,7 @@ protected:
                 for (int i_const = 0; i_const < static_cast<int>(rModelPart.MasterSlaveConstraints().size()); ++i_const) {
                     auto it_const = it_const_begin + i_const;
 
-                    // Detect if the constraint is active or not. If the user did not make any choice the constraint
-                    // It is active by default
+                    // If the constraint is active
                     if(it_const->IsActive()) {
                         it_const->EquationIdVector(slave_ids, master_ids, r_current_process_info);
 
@@ -972,8 +971,7 @@ protected:
             for (int i_const = 0; i_const < number_of_constraints; ++i_const) {
                 auto it_const = rModelPart.MasterSlaveConstraints().begin() + i_const;
 
-                // Detect if the constraint is active or not. If the user did not make any choice the constraint
-                // It is active by default
+                // If the constraint is active
                 if (it_const->IsActive()) {
                     it_const->CalculateLocalSystem(transformation_matrix, constant_vector, r_current_process_info);
                     it_const->EquationIdVector(slave_equation_ids, master_equation_ids, r_current_process_info);

--- a/kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h
+++ b/kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h
@@ -224,7 +224,7 @@ public:
             for (int k = 0; k < nelements; ++k) {
                 auto it_elem = it_elem_begin + k;
 
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // If the element is active
                 if (it_elem->IsActive()) {
                     // Calculate elemental contribution
                     pScheme->CalculateSystemContributions(*it_elem, LHS_Contribution, RHS_Contribution, equation_id, r_current_process_info);
@@ -242,7 +242,7 @@ public:
             for (int k = 0; k < nconditions; ++k) {
                 auto it_cond = it_cond_begin + k;
 
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // If the condition is active
                 if (it_cond->IsActive()) {
                     // Calculate elemental contribution
                     pScheme->CalculateSystemContributions(*it_cond, LHS_Contribution, RHS_Contribution, equation_id, r_current_process_info);
@@ -312,7 +312,7 @@ public:
             for (int k = 0; k < nelements; ++k) {
                 auto it_elem = it_elem_begin + k;
 
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // If the element is active
                 if (it_elem->IsActive()) {
                     // Calculate elemental contribution
                     pScheme->CalculateLHSContribution(*it_elem, LHS_Contribution, equation_id, r_current_process_info);
@@ -326,7 +326,7 @@ public:
             for (int k = 0; k < nconditions; ++k) {
                 auto it_cond = it_cond_begin + k;
 
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // If the condition is active
                 if (it_cond->IsActive()) {
                     // Calculate elemental contribution
                     pScheme->CalculateLHSContribution(*it_cond, LHS_Contribution, equation_id, r_current_process_info);
@@ -390,7 +390,7 @@ public:
             for (int k = 0; k < nelements; ++k) {
                 auto it_elem = it_elem_begin + k;
 
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // If the element is active
                 if (it_elem->IsActive()) {
                     // Calculate elemental contribution
                     pScheme->CalculateLHSContribution(*it_elem, LHS_Contribution, equation_id, r_current_process_info);
@@ -404,7 +404,7 @@ public:
             for (int k = 0; k < nconditions; ++k) {
                 auto it_cond = it_cond_begin + k;
 
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // If the condition is active
                 if (it_cond->IsActive()) {
                     // Calculate elemental contribution
                     pScheme->CalculateLHSContribution(*it_cond, LHS_Contribution, equation_id, r_current_process_info);
@@ -605,7 +605,7 @@ public:
             for (int i = 0; i < nelements; ++i) {
                 auto it_elem = it_elem_begin + i;
 
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // If the element is active
                 if (it_elem->IsActive()) {
                     // Calculate elemental Right Hand Side Contribution
                     pScheme->CalculateRHSContribution(*it_elem, RHS_Contribution, equation_id, r_current_process_info);
@@ -621,7 +621,7 @@ public:
             #pragma omp  for schedule(guided, 512)
             for (int i = 0; i < nconditions; ++i) {
                 auto it_cond = it_cond_begin + i;
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // If the condition is active
                 if (it_cond->IsActive()) {
                     // Calculate elemental contribution
                     pScheme->CalculateRHSContribution(*it_cond, RHS_Contribution, equation_id, r_current_process_info);

--- a/kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h
+++ b/kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h
@@ -868,8 +868,7 @@ protected:
             for (int i_const = 0; i_const < number_of_constraints; ++i_const) {
                 auto it_const = const_begin + i_const;
 
-                // Detect if the constraint is active or not. If the user did not make any choice the constraint
-                // It is active by default
+                // If the constraint is active
                 if(it_const->IsActive()) {
                     it_const->EquationIdVector(ids, second_ids, r_current_process_info);
                     // Slave DoFs
@@ -991,8 +990,7 @@ protected:
         for (int i_const = 0; i_const < number_of_constraints; ++i_const) {
             auto it_const = it_const_begin + i_const;
 
-            // Detect if the constraint is active or not. If the user did not make any choice the constraint
-            // It is active by default
+            // If the constraint is active
             if(it_const->IsActive()) {
                 it_const->EquationIdVector(ids, second_ids, r_current_process_info);
                 for (auto& slave_id : ids) {
@@ -1533,8 +1531,7 @@ private:
             for (int i_const = 0; i_const < number_of_constraints; ++i_const) {
                 auto it_const = it_const_begin + i_const;
 
-                // Detect if the constraint is active or not. If the user did not make any choice the constraint
-                // It is active by default
+                // If the constraint is active
                 if (it_const->IsActive()) {
                     it_const->GetDofList(slave_dof_list, master_dof_list, r_current_process_info);
 
@@ -1788,7 +1785,7 @@ private:
             #pragma omp for schedule(guided, 512) nowait
             for (int i = 0; i<nelements; ++i) {
                 auto it_elem = it_elem_begin + i;
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // If the element is active
                 if (it_elem->IsActive()) {
                     // Calculate elemental contribution
                     pScheme->CalculateSystemContributions(*it_elem, lhs_contribution, rhs_contribution, equation_id, r_current_process_info);
@@ -1804,7 +1801,7 @@ private:
             #pragma omp  for schedule(guided, 512)
             for (int i = 0; i<nconditions; ++i) {
                 auto it_cond = it_cond_begin + i;
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // If the condition is active
                 if (it_cond->IsActive()) {
                     // Calculate elemental contribution
                     pScheme->CalculateSystemContributions(*it_cond, lhs_contribution, rhs_contribution, equation_id, r_current_process_info);
@@ -1853,7 +1850,7 @@ private:
             #pragma omp for schedule(guided, 512) nowait
             for (int i = 0; i<nelements; ++i) {
                 auto it_elem = it_elem_begin + i;
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // If the element is active
                 if (it_elem->IsActive()) {
                     // Calculate elemental Right Hand Side Contribution
                     pScheme->CalculateRHSContribution(*it_elem, rhs_contribution, equation_id, r_current_process_info);
@@ -1869,7 +1866,7 @@ private:
             #pragma omp  for schedule(guided, 512)
             for (int i = 0; i<nconditions; ++i) {
                 auto it_cond = it_cond_begin + i;
-                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // If the condition is active
                 if (it_cond->IsActive()) {
                     // Calculate elemental contribution
                     pScheme->CalculateRHSContribution(*it_cond, rhs_contribution, equation_id, r_current_process_info);
@@ -2101,8 +2098,7 @@ private:
             for (int i_const = 0; i_const < number_of_constraints; ++i_const) {
                 auto it_const = rModelPart.MasterSlaveConstraints().begin() + i_const;
 
-                // Detect if the constraint is active or not. If the user did not make any choice the constraint
-                // It is active by default
+                // If the constraint is active
                 if (it_const->IsActive()) {
                     it_const->CalculateLocalSystem(transformation_matrix, constant_vector, r_current_process_info);
                     it_const->EquationIdVector(slave_equation_id, master_equation_id, r_current_process_info);

--- a/kratos/utilities/entities_utilities.h
+++ b/kratos/utilities/entities_utilities.h
@@ -88,8 +88,7 @@ namespace EntitiesUtilities
         block_for_each(
             r_entities_array,
             [&r_current_process_info](TEntityType& rEntity) {
-                // Detect if the entity is active or not. If the user did not make any choice the entity
-                // It is active by default
+                // If the entity is active
                 if (rEntity.IsActive()) {
                     rEntity.Initialize(r_current_process_info);
                 }

